Game/MyProject/Game/Helper/HelperManager.cpp

```diff
@@ -0,0 +1,548 @@
+    vertex3.position=points[6];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[7];
+    vertex2.position=points[1];
+    vertex3.position=points[8];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    
+    vertex1.position=points[3];
+    vertex2.position=points[9];
+    vertex3.position=points[4];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[3];
+    vertex2.position=points[4];
+    vertex3.position=points[2];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[3];
+    vertex2.position=points[2];
+    vertex3.position=points[6];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[3];
+    vertex2.position=points[6];
+    vertex3.position=points[8];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[3];
+    vertex2.position=points[8];
+    vertex3.position=points[9];
+
+    
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    
+    vertex1.position=points[4];
+    vertex2.position=points[9];
+    vertex3.position=points[5];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[2];
+    vertex2.position=points[4];
+    vertex3.position=points[11];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[6];
+    vertex2.position=points[2];
+    vertex3.position=points[10];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[8];
+    vertex2.position=points[6];
+    vertex3.position=points[7];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[9];
+    vertex2.position=points[8];
+    vertex3.position=points[1];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    vertex1.position=points[0];
+    vertex2.position=points[11];
+    vertex3.position=points[5];
+    
+    vertices.push_back(vertex1);
+    vertices.push_back(vertex2);
+    vertices.push_back(vertex3);
+    
+    std::vector<IvTNPVertex> vertices2;
+    vertices2=RefineTriangles(vertices, 4);
+    sphereMesh->setVertexBuffer(vertices2, format);
+    
+    std::vector<unsigned int> indices;
+    for(int i=0;i<vertices2.size();i++)
+        indices.push_back(i);
```

Comment:
again, very bad use of index buffers, just like the box one.
this whole method looks weird with ~150lines of specific vertex pushing.
you should be able to generate the needed vertex positions in a loop using polar coordinates.